pkg/controller/nodes/dynamic/handler.go

```diff
@@ -377,8 +379,19 @@ func (d dynamicNodeTaskNodeHandler) buildContextualDynamicWorkflow(ctx context.C
 		return nil, true, err
 	}
 
-	// TODO: This will currently fail if the WF references any launch plans
-	closure, err = compiler.CompileWorkflow(wf, djSpec.Subworkflows, compiledTasks, []common2.InterfaceProvider{})
+	// Get the requirements, that is, a list of all the task IDs and the launch plan IDs that will be called as part of this dynamic task.
+	// The definition of these will need to be fetched from Admin (in order to get the interface).
+	requirements, err := compiler.GetRequirements(wf, djSpec.Subworkflows)
+	if err != nil {
+		return nil, true, err
+	}
+
+	launchPlanInterfaces, err := d.getLaunchPlanInterfaces(ctx, requirements.GetRequiredLaunchPlanIds())
+	if err != nil {
+		return nil, true, err
+	}
+
```

Comment:
Mind adding a TODO to get the tasks we don't already have from Admin as well...

Reply:
Yup.